### DISCUSSION TITLE
Cloudprober log level

### DISCUSSION THREAD

**Discussion by markoposavec** [cloudprober/cloudprober#792] [Q&A]
Hi,
Is there a way to set the logging level for Cloudprober? I would like to disable info logs, but from what I see in code it is not possible. Are there plans in the future to have an option for selecting log level?

Thanks a lot,
Marko

**Comment by markoposavec** [comment]
In particular, I've created external probe and I'm passing username and password as arguments to the program and this line logs the whole command that is executed:
https://github.com/cloudprober/cloudprober/blob/master/probes/external/external.go#L342
which I would like to avoid.
Also I've noticed that in HTTP requests that fail whole URL is logged, and in some probes we send sensitive data in query parameters for some old API's that we cannot change :cry: 
Only way for us to send the query parameters is to write them in relative url part of HTTP probe. Would it make sense to create the query parameters field in HTTP probes that wouldn't be logged or just omit/replace the query parameters in logs (if some flag is set)?

**Comment by manugarg** [comment]
Hello Mark,

There is no such option as of now, but we should implement such an option. I've created an issue from it:
https://github.com/cloudprober/cloudprober/issues/798

I'll try to add this option soonish.
